cli/commands/catalog/module/doc.go

```diff
@@ -0,0 +1,307 @@
+			// header
+			regexp.MustCompile(`(?m)^\#{1,6}\s*([^#]+)\s*(\#{1,6})?$`),
+			// horizontal rule
+			regexp.MustCompile(`^[-\*_]{3,}\s*$`),
+		}
+
+	case adocExt:
+		doc.tagRegs[tagH1Block] = regexp.MustCompile(`(?:^|\n)\={1}\s([\S\s]+?)(?:[\r\n]+\=|[\r\n]*$)`)
+		doc.tagRegs[tagH2Block] = regexp.MustCompile(`(?:^|\n)\={2}\s([\S\s]+?)(?:[\r\n]+\=|[\r\n]*$)`)
+		doc.tagStripRegs = DocRegs{
+			// html
+			regexp.MustCompile("<(.*?)>"),
+			// ifdef endif
+			regexp.MustCompile(`ifdef::env-github\[\][\S\s]*?endif::\[\]`),
+			// comment
+			regexp.MustCompile(`(?m)^/{2}.*$`),
+			// ex. :name:value
+			regexp.MustCompile(`(?m)^:[-!\w]+:.*$`),
+			// ex. toc::[]
+			regexp.MustCompile(`\w+::\[.*?\]`),
+			// bold
+			regexp.MustCompile(`\*\*([^\s][^*]+[^\s])\*\*`),
+			regexp.MustCompile(`\*([^\s][^*]+[^\s])\*`),
+			// italic
+			regexp.MustCompile(`_{1,2}([^\s][^_]+[^\s])_{1,2}`),
+			// image
+			regexp.MustCompile(`image:[^\]]+]`),
+			// link
+			regexp.MustCompile(`(?:link|https):[\S\s]+?\[([\S\s]+?)\]`),
+			// header
+			regexp.MustCompile(`(?m)^\={1,6}\s*([^=]+)\s*(\={1,6})?$`),
+			// multiple line break
+			regexp.MustCompile(`((?:\r\n?|\n){2})(?:\r\n?|\n)*`),
+		}
+	}
+
+	return doc
+}
+
+func FindDoc(dir string) (*Doc, error) {
+	var filePath, fileExt string
+
+	files, err := os.ReadDir(dir)
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+
+	for _, file := range files {
+		if file.IsDir() {
+			continue
+		}
+
+		for _, readmeFile := range docFiles {
+			if strings.EqualFold(readmeFile, file.Name()) {
+				filePath = filepath.Join(dir, file.Name())
+				fileExt = filepath.Ext(filePath)
+				break
+			}
+		}
+
+		// `md` files have priority over `adoc` files
+		if strings.EqualFold(fileExt, mdExt) {
+			break
+		}
+	}
+
+	if filePath == "" {
+		return &Doc{}, nil
+	}
+
+	contentByte, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, errors.WithStackTrace(err)
+	}
+	rawContent := string(contentByte)
+
+	return newDoc(rawContent, fileExt), nil
+}
+
+func (doc *Doc) Title() string {
+	if title := doc.parseFrontmatter(docTitle); title != "" {
+		return title
+	}
+
+	return doc.parseTag(docTitle)
+}
+
+func (doc *Doc) Description(maxLenght int) string {
+	desc := doc.parseFrontmatter(docDescription)
+
+	if desc == "" {
+		desc = doc.parseTag(docDescription)
+	}
+
+	if maxLenght == 0 {
+		return desc
+	}
+
+	var (
+		sentences = strings.Split(desc, ".")
+		symbols   int
+	)
+
+	for i, sentence := range sentences {
+		symbols += len(sentence)
+
+		if symbols > maxLenght {
+			if i == 0 {
+				desc = sentence
+			} else {
+				desc = strings.Join(sentences[:i], ".")
+			}
+
+			desc += "."
+			break
+		}
+	}
+
+	return desc
+}
+
+func (doc *Doc) Content(stripTags bool) string {
+	if !stripTags {
+		return doc.rawContent
+	}
+
+	return doc.parseTag(docContent)
+}
+
+func (doc *Doc) IsMarkDown() bool {
+	return doc.fileExt == mdExt
+}
+
+func (doc *Doc) parseFrontmatter(key docDataKey) string {
```

Comment:
NIT: some comments in this file explaining what we're doing (i.e., that some modules have Markdown files with frontmatter, which we use as the preferred title/description source) may help!

Reply:
Oh, sorry, I should have commented these funcs.  I will do it in the next task in order not to again bother you with a request for a review.